### PR TITLE
Provider: fixes in Provider, Provider Docs and OnNavigation-related component Docs

### PR DESCRIPTION
**Pull request by AlbertCarreras**
Fix for
- Provider examples (cleanup)
- Provider onNavigation descriptions (improved texts)
- Stylistic writing issues (following Pinterest writing guidelines)
- OnNavigation types
- Single/Double Quotes (follow Pinboard guidelines)
- Props table for consumers (standardized text)
- Added new parameter (target) being passed to onNavigation function on Provider

### CODE REVIEW

File: docs/src/Provider.doc.js [hunk 1]
```diff
@@ -296,10 +297,11 @@ function OnNavigation() {
 `}
       />
       <MainSection.Card
+        title="Example above: Dropdown"
         cardSize="lg"
         defaultCode={`
 function OnNavigation() {
-  const [clientOnNavigationMode, setClientOnNavigationMode] = React.useState(true);
+  const [clientOnNavigationMode, setClientOnNavigationMode] = React.useState("link");
```
Comment (dangerismycat) [inline docs/src/Provider.doc.js:304]: nit: single quotes

File: docs/src/Provider.doc.js [hunk 2]
```diff
@@ -111,41 +113,39 @@ function OnNavigation() {
...
-            onChange={() => setClientOnNavigationMode(!clientOnNavigationMode)}
-            id="disable-buttons"
-            switched={clientOnNavigationMode}
+        <Flex direction="column" gap={2}>
+          <Text>Navigation type:</Text>
+          <RadioButton
+            checked={clientOnNavigationMode === "link"}
+            id="link"
+            label="Default Link"
+            name="navigation"
```
Comment (ayeshakmaz) [inline docs/src/Provider.doc.js:136]: Ahh each set of RadioButtons needs a unique name (can't click the 2nd and 3rd sets right now)

**Comment by dangerismycat** [comment]
Is there an example where we show an onNavigationOptions value of a Node?

Also, this text needs to be broken up into 2-3 paragraphs:
Screen Shot 2021-02-05 at 3 51 35 PM

File: docs/src/Provider.doc.js [hunk 3]
```diff
@@ -191,10 +191,11 @@ function OnNavigation() {
 `}
       />
       <MainSection.Card
+        title="Examples above from top to bottom: (1) Callout, (2) Upsell, (3) ActivationCard"
```
Comment (ayeshakmaz) [inline docs/src/Provider.doc.js:194]: Titles are great, thank you! Side note: I'm gonna that with Annie about this styling though, I'm thinking for large examples the title needs to stay above

File: docs/src/Dropdown.doc.js [hunk 4]
```diff
@@ -212,7 +212,8 @@ card(
         type: '({ [string]: Node | ({| +event: SyntheticEvent<> |}) => void }) => void',
         description: [
           'Pass custom props to onNavigation.',
-          'onNavigationOptions works in pair with a Provider. See Provider for examples.',
+          'Pass custom props to onNavigation. See Provider for examples.',
```
Comment (ayeshakmaz) [inline docs/src/Dropdown.doc.js:215]: Duplicate text here

File: docs/src/Provider.doc.js (hunk 2 above)
Comment (AlbertCarreras) [reply docs/src/Provider.doc.js:136]: DONE

File: docs/src/Provider.doc.js (hunk 1 above)
Comment (AlbertCarreras) [reply docs/src/Provider.doc.js]: DONE

File: docs/src/Provider.doc.js (hunk 3 above)
Comment (AlbertCarreras) [reply docs/src/Provider.doc.js:194]: agreed

File: docs/src/Dropdown.doc.js (hunk 4 above)
Comment (AlbertCarreras) [reply docs/src/Dropdown.doc.js:214]: DONE

**Comment by AlbertCarreras** [comment]
> Is there an example where we show an onNavigationOptions value of a Node?
> 
> Also, this text needs to be broken up into 2-3 paragraphs:
> Screen Shot 2021-02-05 at 3 51 35 PM

 https://flow.org/en/docs/react/types/#toc-react-node React.Node can be null, a boolean, a number, a string, a React element, or an array of any of those types recursively.

File: docs/src/Provider.doc.js (hunk 2 above)
Comment (ayeshakmaz) [reply docs/src/Provider.doc.js:136]: Sorry just noticed it - all the value="dogs" should be removed 😛

Comment (AlbertCarreras) [reply docs/src/Provider.doc.js:136]: DONE, yeah noticed it too

File: docs/src/Provider.doc.js [hunk 5]
```diff
@@ -93,59 +94,68 @@ function Example(props) {
...
       event.nativeEvent.preventDefault();
       // eslint-disable-next-line no-alert
-      alert("Disabled link. Opening help.pinterest.com instead.");
-      window.open("https://help.pinterest.com", "_blank");
+      alert('Disabled link. Opening help.pinterest.com instead.');
+      window.open('https://help.pinterest.com', '_blank');
     }
 
-    return onNavigationOptions && onNavigationOptions.navigationMode === "client"
+    return onNavigationOptions && onNavigationOptions.navigationMode === 'navigation'
```
Comment (ayeshakmaz) [inline docs/src/Provider.doc.js:126]: Think these should each be 'onNavigation' (or the radios pass in 'navigation'

File: docs/src/Provider.doc.js [hunk 6]
```diff
@@ -93,59 +94,68 @@ function Example(props) {
...
+      title="Custom navigation context"
+      description={`
+        These examples illustrate a custom implementation of \`onNavigation\` context to control the link functionality of Gestalt components externally.
+
+        This example has 4 relevant parts: a Provider, an \`onNavigation\` high-order function, consumer components, and \`onNavigationOptions\` props.
+
+        The top Provider passes the custom \`onNavigation\` function to consumer components. Then, \`onNavigation\` returns a function that gets called during the \`onClick\` event handler.
+
+        The \`onNavigation\` function can contain complex logic including React hooks to perform side effects. In this case, \`onNavigation\` is used to disable the default link, show an alert message, and open a different URL in a new window.
+
```
Comment (ayeshakmaz) [inline docs/src/Provider.doc.js:106]: Maybe: 

The \`onNavigation\` function can contain complex logic, including React hooks, to perform side effects. In this case, \`onNavigation\` is used to 
  - disable the default Link behavior
  - show an alert message
  - open a different URL in a new window

File: docs/src/Provider.doc.js [hunk 7]
```diff
@@ -93,59 +94,68 @@ function Example(props) {
...
+        Finally, the \`onNavigationOptions\` prop provides a flexible API. The \`onNavigationOptions\` is passed as an argument to the \`onNavigation\` function providing external control to the logic inside the function. In this case, \`onNavigationOptions\` toggles a \`navigationMode\` key between \`default\` and \`navigation\` values. Then, inside the \`onNavigation\` function, the \`navigationMode\` value toggles between the default link behaviour and the custom navigation context one. Other uses could be accessing \`event.stopPropagation\`.
```
Comment (ayeshakmaz) [inline docs/src/Provider.doc.js:107]: Maybe:

Finally, the \`onNavigationOptions\` prop provides a flexible API. The \`onNavigationOptions\` prop is passed as an argument to the \`onNavigation\` function, providing external control to the logic inside the function. In this case, \`onNavigationOptions\` toggles a \`navigationMode\` key between \`default\` and \`navigation\` values. Then, inside the \`onNavigation\` function, the \`navigationMode\` value toggles between the default link behavior and the custom navigation context behavior. Other uses for this level of customization could be accessing \`event.stopPropagation\`.

File: docs/src/Provider.doc.js (hunk 6 above)
Comment (AlbertCarreras) [reply docs/src/Provider.doc.js:106]: DONE

File: docs/src/Provider.doc.js (hunk 7 above)
Comment (AlbertCarreras) [reply docs/src/Provider.doc.js]: DONE

File: docs/src/Provider.doc.js (hunk 5 above)
Comment (AlbertCarreras) [reply docs/src/Provider.doc.js:129]: DONE

File: docs/src/Provider.doc.js [hunk 8]
```diff
@@ -93,59 +94,71 @@ function Example(props) {
...
+      title="Custom navigation context"
+      description={`
+        These examples illustrate a custom implementation of \`onNavigation\` context to control the link functionality of Gestalt components externally.
+
+        This example has four relevant parts: a Provider, an \`onNavigation\` high-order function, consumer components and \`onNavigationOptions\` props.
+
+        The top Provider passes the custom \`onNavigation\` function to consumer components which execute it. Then, \`onNavigation\` returns a function that gets called during the \`onClick\` event handler.
+
+        The \`onNavigation\` function can contain complex logic, including React hooks, to perform side effects. It also takes named arguments: \`href\`, \`onNavigationOptions\` and \`target\`. In this case, \`onNavigation\` executes the following actions:
+          - Disable the default Link behavior
```
Comment (ayeshakmaz) [inline docs/src/Provider.doc.js:106]: One last tiny nit: "In the examples below," instead of "In this case,"

**Review by dangerismycat — APPROVED** [review]

**Review by ayeshakmaz — APPROVED** [review]
Love it!

File: docs/src/Provider.doc.js (hunk 8 above)
Comment (AlbertCarreras) [reply docs/src/Provider.doc.js:106]: DONE

**Pull request merged**